core/trino-main/src/main/java/io/trino/sql/gen/ExpressionCompiler.java

```diff
@@ -14,98 +14,45 @@
 import io.trino.sql.gen.columnar.DynamicPageFilter;
 import io.trino.sql.gen.columnar.FilterEvaluator;
 import io.trino.sql.gen.columnar.PageFilterEvaluator;
 import io.trino.sql.relational.RowExpression;
-import org.weakref.jmx.Managed;
-import org.weakref.jmx.Nested;
 
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 import java.util.OptionalInt;
 import java.util.function.Function;
 import java.util.function.Supplier;
 
-import static com.google.common.base.MoreObjects.toStringHelper;
-import static com.google.common.base.Throwables.throwIfInstanceOf;
 import static com.google.common.collect.ImmutableList.toImmutableList;
-import static io.airlift.bytecode.Access.FINAL;
-import static io.airlift.bytecode.Access.PUBLIC;
-import static io.airlift.bytecode.Access.a;
-import static io.airlift.bytecode.ParameterizedType.type;
-import static io.trino.cache.SafeCaches.buildNonEvictableCache;
-import static io.trino.spi.type.BooleanType.BOOLEAN;
-import static io.trino.sql.gen.BytecodeUtils.invoke;
 import static io.trino.sql.gen.columnar.FilterEvaluator.createColumnarFilterEvaluator;
-import static io.trino.sql.relational.Expressions.constant;
-import static io.trino.util.CompilerUtils.defineClass;
-import static io.trino.util.CompilerUtils.makeClassName;
 import static java.util.Objects.requireNonNull;
 
 public class ExpressionCompiler
 {
+    private final CursorProcessorCompiler cursorProcessorCompiler;
     private final PageFunctionCompiler pageFunctionCompiler;
     private final ColumnarFilterCompiler columnarFilterCompiler;
-    private final NonEvictableLoadingCache<CacheKey, Class<? extends CursorProcessor>> cursorProcessors;
-    private final CacheStatsMBean cacheStatsMBean;
 
     @Inject
-    public ExpressionCompiler(FunctionManager functionManager, PageFunctionCompiler pageFunctionCompiler, ColumnarFilterCompiler columnarFilterCompiler)
+    public ExpressionCompiler(CursorProcessorCompiler cursorProcessorCompiler, PageFunctionCompiler pageFunctionCompiler, ColumnarFilterCompiler columnarFilterCompiler)
     {
-        requireNonNull(functionManager, "functionManager is null");
+        this.cursorProcessorCompiler = requireNonNull(cursorProcessorCompiler, "cursorProcessorCompiler is null");
         this.pageFunctionCompiler = requireNonNull(pageFunctionCompiler, "pageFunctionCompiler is null");
         this.columnarFilterCompiler = requireNonNull(columnarFilterCompiler, "columnarFilterCompiler is null");
-        this.cursorProcessors = buildNonEvictableCache(CacheBuilder.newBuilder()
-                        .recordStats()
-                        .maximumSize(1000),
-                CacheLoader.from(key -> compile(key.getFilter(), key.getProjections(), new CursorProcessorCompiler(functionManager), CursorProcessor.class)));
-        this.cacheStatsMBean = new CacheStatsMBean(cursorProcessors);
     }
 
-    @Managed
-    @Nested
-    public CacheStatsMBean getCursorProcessorCache()
```

Comment:
This is a breaking change because the location of this metric has changed. Are we okay with this, or should I leave the metric in its previous location? I moved it to align with the other compilers used in the ExpressionCompiler.

Reply:
It's ok. I don't think that we need to call this out in the docs.